app/models/manageiq/providers/ansible_tower/shared/inventory/collector/configuration_script_source.rb

```diff
@@ -4,7 +4,7 @@ def connection
   end
 
   def projects
-    target.refresh_in_provider
+    target.class.refresh_in_provider(project)
```

Comment:
I hate being a jerk about this, but I think the instance method should remain, even if there *needs* to be a class method available for calling `refresh_in_provider` on objects not yet in the MIQ DB.

I would see it as:

```ruby
module Shared::ConfigurationScriptSource
  module ClassMethods
    def refresh_in_provider(project)
      # ... full impl
    end
  end

  def refresh_in_provider
    with_provider_object { |project| self.class.refresh_in_provider(project) }
  end
end
```

Then, callers don't need to do all the `class` dereferencing on instances and then have to know that they need to get the underlying provider project reference to pass into the class method.

And, then this line would just revert to the original source.